### PR TITLE
Fix instance splitter issue with short time series

### PR DESCRIPTION
**Pull request by lostella**
*Description of changes:* Some checks are added in the `InstanceSplitter` class for the case where `pick_incomplete=False`. This would otherwise cause problems downstream, when the sampler would to sample from an interval that has inverted bounds.

Further checks and typing cleanup are added.

Opening for feedback, trying to include a small test for this.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by lostella** [comment]
Polished the code, updated the tests, added test cases with `pick_incomplete=True` (I verified that these raise the issue on `master`)

File: test/test_transform.py [hunk 1]
```diff
@@ -204,10 +209,25 @@ def test_InstanceSplitter(start, target, is_train):
...
 
     if is_train:
-        assert len(out) == max(0, len(target) - pred_length + 1)
+        assert len(out) == max(
+            0,
+            len(target)
+            - pred_length
+            + 1
+            - (0 if pick_incomplete else train_length),
+        )
```
Comment (lostella) [inline test/test_transform.py]: Sometimes I really don't understand black

File: src/gluonts/transform/split.py [hunk 2]
```diff
@@ -156,30 +156,37 @@ def flatmap_transform(
...
+            # mask the loss.
+            sampled_indices = (
+                np.array([], dtype=int)
+                if len_target < minimum_length
+                else self.train_sampler(target, *sampling_bounds)
+            )
         else:
-            sampling_indices = [len_target]
-        for i in sampling_indices:
+            assert self.pick_incomplete or len_target >= minimum_length
```
Comment (vafl) [inline src/gluonts/transform/split.py]: For inference we don't need the future, so I think this should be

```python
asser self.pick_incomplete or len_target >= self.past_length
```

File: test/test_transform.py [hunk 3]
```diff
@@ -204,10 +209,25 @@ def test_InstanceSplitter(start, target, is_train):
         "some_other_col": "ABC",
     }
 
-    out = list(t.flatmap_transform(data, is_train=is_train))
+    if (
+        not is_train
+        and not pick_incomplete
+        and len(target) < train_length + pred_length
```
Comment (vafl) [inline test/test_transform.py]: and correspondingly here 

```
len(target) < train_length
```

Comment (lostella) [reply test/test_transform.py]: good catch!

**Pull request merged**